harness/determined/cli/command.py

```diff
@@ -282,15 +284,38 @@ def config(args: Namespace) -> None:
     print(render.format_object_as_yaml(res_json["config"]))
 
 
-def _set_nested_config(config: Dict[str, Any], key_path: List[str], value: Any) -> Dict[str, Any]:
-    current = config
-    for key in key_path[:-1]:
-        current = current.setdefault(key, {})
-    current[key_path[-1]] = value
-    return config
-
+# Convert config overrides in dot notation to json path
+# For example the caller of this func will break down the config override
+#   --config slurm.sbatch_args=[--time=05:00]
+# into key slurm.sbatch_args, value [--time=05:00]
+# And this func will return string in json form:
+# 'slurm': {'sbatch_args': ['--time=05:00']}
+def _dot_to_json(key: Any, value: Any) -> Any:
+    json_output: Any = {}
+    path = key.split(".")
+    target = reduce(lambda d, k: d.setdefault(k, {}), path[:-1], json_output)
+    target[path[-1]] = value
+    return json_output
+
+
+# A recursive function to replace any value val in a Dict with a new value rval
+def _replace_value(data_dict: Dict[str, Any], val: Any, rval: Any) -> None:
+    for key in data_dict.keys():
+        if data_dict[key] == val:
+            data_dict[key] = rval
+        elif type(data_dict[key]) is dict:
+            _replace_value(data_dict[key], val, rval)
+
+
+def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) -> Dict[str, Any]:
+    if any(True for _ in overrides):
```

Comment:
this is a bit of a problem if `overrides` is not a list but one-time iterable like a generator: this for loop will consume the first element, and the iteration in line 319 will start iterating from the second element.

I took a quick glance and it seems all the times we pass overrides, it's a list. so we should probably change the typing here to be:

```suggestion
def parse_config_overrides(config: Dict[str, Any], overrides: List[str]) -> Dict[str, Any]:
    if any(True for _ in overrides):
```

Reply:
okay

Reply:
Looks like changing overrides: Iterable to List is no longer needed after the change below